docs/src/covariance_parameters/general.md

```diff
@@ -0,0 +1,15 @@
+# General 
+
+```@meta
+CurrentModule = MarkovKernels
+```
+
+MarkovKernels.jl supports covariance matrices of the following represenration:
+
+```julia
+const RSym{T,S} = Symmetric{T,S} where {T<:Real,S}
+const CHerm{T,S} = Hermitian{T,S} where {T<:Complex,S}
+const SelfAdjoint{T,S} = Union{RSym{T,S},CHerm{T,S}} where {T,S}
+
+const CovarianceParameter{T} = Union{SelfAdjoint{T},Factorization{T}}
```

Comment:
So this means that to be a valid covariance parameter, it's either `SelfAdjoint` or _any_ `Factorization`? Why is the latter a reasonable thing to do? If I understand this correctly, this allows e.g. QR decompositions but forbids regular matrices, which would be a bit surprising to me.

Reply:
Roughly speaking the significance of ```CovarianceParameter``` is the following: 

1. It indicates, which types have implemented the, let's call it for now, "covariance interface".
2. It is used to define ```AffineNormalKernel``` by :
```julia
const AffineNormalKernel{T} =
    NormalKernel{<:AbstractAffineMap{T},<:CovarianceParameter{T}} where {T}
```

Now this  means that it impossible opt into making your type a ```CovarianceParameter```. So that you would have to add methods for types ```NormalKernel{<:AbstractAffineMap,<:MyCovarianceType}```. 

My solution to this would be to do the following: 

```julia
abstract type AbstractAffineNormalKernel <: AbstractNormalKernel end

struct AffineNormalKernel{A,B} <: AbstractAffineNormalKernel 
affine_map::A
covariance::B
end
```

So that all operations on combinations of ```Dirac```, ```DiracKernel```, ```Normal```, ```AffineNormalKernel``` 
"just work" as long as their supplied with vectors that behave as ```AbstractVector```, affine maps that behave like ```AbstractAffineMap```, and covariances that behave ```CovarianceParameter``` (in the sense that they implement the necessary methods).

However, I was planning this for another PR but if you want it in this one I could probably arrange that too.

Reply:
I think putting it into another PR is also fine. Also, I'm not quite 100% sure what functionality was there before, so many of my questions are really just for understanding, not necessarily because the actual PRs are lacking something. As long as this PR doesn't add new restrictions that were not there before I'm good with it, and then loosening those could indeed be better done in a separate PR :+1:

Reply:
Great. Lets save it for later then.